File: BankImport/BankImport/Slf.Dms.Service.BankImport.Slf.Dms.Service.BankImport/ImportMainForm.vb
Imports Slf.Dms.Service
Imports System
Imports System.Collections
Imports System.ComponentModel
Imports System.Diagnostics
Imports System.Drawing
Imports System.Windows.Forms

Namespace Slf.Dms.Service.BankImport.Slf.Dms.Service.BankImport
    Public Class ImportMainForm
        Inherits Form
        ' Methods
        Public Sub New()
            Dim list As ArrayList = ImportMainForm.__ENCList_
            SyncLock list
                ImportMainForm.__ENCList_.Add(New WeakReference(Me))
            End SyncLock
            Me.InitializeComponent()
        End Sub

        Private Sub browseButton_Click(ByVal sender As Object, ByVal e As EventArgs)
            Using dialog As OpenFileDialog = New OpenFileDialog
                dialog.InitialDirectory = ImportMain.FileLocation
                If (dialog.ShowDialog(Me) = DialogResult.OK) Then
                    Me.dataFileTextBox.Text = dialog.FileName
                End If
            End Using
        End Sub

        Protected Overrides Sub Dispose(ByVal disposing As Boolean)
            If (IIf((disposing AndAlso (Not Me.components Is Nothing)), 1, 0) <> 0) Then
                Me.components.Dispose()
            End If
            MyBase.Dispose(disposing)
        End Sub

        Private Sub importButton_Click(ByVal sender As Object, ByVal e As EventArgs)
            ImportMain.ImportFile(Me.dataFileTextBox.Text, Me.databaseTextBox.Text)
        End Sub

        Private Sub ImportMainForm_Load(ByVal sender As Object, ByVal e As EventArgs)
            Me.databaseTextBox.Text = ImportMain.ConnectionString
            Trace.Listeners.Add(New Drg.Util.TextBoxTraceListener(Me.logTextBox))
        End Sub

        Private Sub InitializeComponent()
            Me.label3 = New Label
            Me.label2 = New Label
            Me.label1 = New Label
            Me.importButton = New Button
            Me.logTextBox = New TextBox
            Me.databaseTextBox = New TextBox
            Me.browseButton = New Button
            Me.dataFileTextBox = New TextBox
            Me.runStoredProcCheckBox = New CheckBox
            MyBase.SuspendLayout()
            Me.label3.AutoSize = True
            Dim point As New Point(&H10, &H88)
            Me.label3.Location = point
            Me.label3.Name = "label3"
            Dim size As New Size(&H1A, &H10)
            Me.label3.Size = size
            Me.label3.TabIndex = 5
            Me.label3.Text = "&Log:"
            Me.label2.AutoSize = True
            point = New Point(&H10, &H30)
            Me.label2.Location = point
            Me.label2.Name = "label2"
            size = New Size(&H61, &H10)
            Me.label2.Size = size
            Me.label2.TabIndex = 2
            Me.label2.Text = "&Connection String:"
            Me.label1.AutoSize = True
            point = New Point(&H10, &H10)
            Me.label1.Location = point
            Me.label1.Name = "label1"
            size = New Size(&H37, &H10)
            Me.label1.Size = size
            Me.label1.TabIndex = 0
            Me.label1.Text = "Bank &File:"
            point = New Point(&HC0, &H70)
            Me.importButton.Location = point
            Me.importButton.Name = "importButton"
            size = New Size(&H4F, &H17)
            Me.importButton.Size = size
            Me.importButton.TabIndex = 4
            Me.importButton.Text = "&Import"
            AddHandler Me.importButton.Click, New EventHandler(AddressOf Me.importButton_Click)
            Me.logTextBox.Anchor = (AnchorStyles.Right Or (AnchorStyles.Left Or (AnchorStyles.Bottom Or AnchorStyles.Top)))
            Me.logTextBox.BackColor = SystemColors.Window
            point = New Point(&H10, &H98)
            Me.logTextBox.Location = point
            Me.logTextBox.Multiline = True
            Me.logTextBox.Name = "logTextBox"
            Me.logTextBox.ReadOnly = True
            Me.logTextBox.ScrollBars = ScrollBars.Both
            size = New Size(420, &HD1)
            Me.logTextBox.Size = size
            Me.logTextBox.TabIndex = 6
            Me.logTextBox.Text = ""
            Me.databaseTextBox.Anchor = (AnchorStyles.Right Or (AnchorStyles.Left Or AnchorStyles.Top))
            point = New Point(120, &H30)
            Me.databaseTextBox.Location = point
            Me.databaseTextBox.Name = "databaseTextBox"
            size = New Size(&H13C, 20)
            Me.databaseTextBox.Size = size
            Me.databaseTextBox.TabIndex = 3
            Me.databaseTextBox.Text = ""
            Me.browseButton.Anchor = (AnchorStyles.Right Or AnchorStyles.Top)
            point = New Point(&H198, &H10)
            Me.browseButton.Location = point
            Me.browseButton.Name = "browseButton"
            size = New Size(&H1C, &H17)
            Me.browseButton.Size = size
            Me.browseButton.TabIndex = 9
            Me.browseButton.Text = "..."
            AddHandler Me.browseButton.Click, New EventHandler(AddressOf Me.browseButton_Click)
            Me.dataFileTextBox.Anchor = (AnchorStyles.Right Or (AnchorStyles.Left Or AnchorStyles.Top))
            point = New Point(120, &H10)
            Me.dataFileTextBox.Location = point
            Me.dataFileTextBox.Name = "dataFileTextBox"
            size = New Size(&H11C, 20)
            Me.dataFileTextBox.Size = size
            Me.dataFileTextBox.TabIndex = 1
            Me.dataFileTextBox.Text = ""
            Me.runStoredProcCheckBox.Checked = True
            Me.runStoredProcCheckBox.CheckState = CheckState.Checked
            point = New Point(120, &H48)
            Me.runStoredProcCheckBox.Location = point
            Me.runStoredProcCheckBox.Name = "runStoredProcCheckBox"
            size = New Size(&H138, &H18)
            Me.runStoredProcCheckBox.Size = size
            Me.runStoredProcCheckBox.TabIndex = 10
            Me.runStoredProcCheckBox.Text = "&Update transactions"
            size = New Size(5, 13)
            Me.AutoScaleBaseSize = size
            size = New Size(&H1C4, &H176)
            MyBase.ClientSize = size
            MyBase.Controls.Add(Me.runStoredProcCheckBox)
            MyBase.Controls.Add(Me.label3)
            MyBase.Controls.Add(Me.label2)
            MyBase.Controls.Add(Me.label1)
            MyBase.Controls.Add(Me.logTextBox)
            MyBase.Controls.Add(Me.databaseTextBox)
            MyBase.Controls.Add(Me.dataFileTextBox)
            MyBase.Controls.Add(Me.importButton)
            MyBase.Controls.Add(Me.browseButton)
            MyBase.Name = "ImportMainForm"
            Me.Text = "Manual Import"
            AddHandler MyBase.Load, New EventHandler(AddressOf Me.ImportMainForm_Load)
            MyBase.ResumeLayout(False)
        End Sub


        ' Fields
        Private Shared __ENCList_ As ArrayList = New ArrayList
        Private browseButton As Button
        Private components As Container
        Private databaseTextBox As TextBox
        Private dataFileTextBox As TextBox
        Private importButton As Button
        Private label1 As Label
        Private label2 As Label
        Private label3 As Label
        Private logTextBox As TextBox
        Private runStoredProcCheckBox As CheckBox
    End Class
End Namespace

File: Slf.Dms.Service.BankImport/ImportMainForm.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;

using Drg.Util;

namespace Slf.Dms.Service.BankImport
{
	/// <summary>
	/// Summary description for Form1.
	/// </summary>
	public class ImportMainForm : System.Windows.Forms.Form
	{
		private System.Windows.Forms.Label label3;
		private System.Windows.Forms.Label label2;
		private System.Windows.Forms.Label label1;
		private System.Windows.Forms.Button importButton;
		private System.Windows.Forms.TextBox logTextBox;
		private System.Windows.Forms.TextBox databaseTextBox;
		private System.Windows.Forms.Button browseButton;
		private System.Windows.Forms.TextBox dataFileTextBox;
		private System.Windows.Forms.CheckBox runStoredProcCheckBox;
		/// <summary>
		/// Required designer variable.
		/// </summary>
		private System.ComponentModel.Container components = null;

		public ImportMainForm()
		{
			//
			// Required for Windows Form Designer support
			//
			InitializeComponent();

			//
			// TODO: Add any constructor code after InitializeComponent call
			//
		}

		/// <summary>
		/// Clean up any resources being used.
		/// </summary>
		protected override void Dispose( bool disposing )
		{
			if( disposing )
			{
				if (components != null) 
				{
					components.Dispose();
				}
			}
			base.Dispose( disposing );
		}

		#region Windows Form Designer generated code
		/// <summary>
		/// Required method for Designer support - do not modify
		/// the contents of this method with the code editor.
		/// </summary>
		private void InitializeComponent()
		{
			this.label3 = new System.Windows.Forms.Label();
			this.label2 = new System.Windows.Forms.Label();
			this.label1 = new System.Windows.Forms.Label();
			this.importButton = new System.Windows.Forms.Button();
			this.logTextBox = new System.Windows.Forms.TextBox();
			this.databaseTextBox = new System.Windows.Forms.TextBox();
			this.browseButton = new System.Windows.Forms.Button();
			this.dataFileTextBox = new System.Windows.Forms.TextBox();
			this.runStoredProcCheckBox = new System.Windows.Forms.CheckBox();
			this.SuspendLayout();
			// 
			// label3
			// 
			this.label3.AutoSize = true;
			this.label3.Location = new System.Drawing.Point(16, 136);
			this.label3.Name = "label3";
			this.label3.Size = new System.Drawing.Size(26, 16);
			this.label3.TabIndex = 5;
			this.label3.Text = "&Log:";
			// 
			// label2
			// 
			this.label2.AutoSize = true;
			this.label2.Location = new System.Drawing.Point(16, 48);
			this.label2.Name = "label2";
			this.label2.Size = new System.Drawing.Size(97, 16);
			this.label2.TabIndex = 2;
			this.label2.Text = "&Connection String:";
			// 
			// label1
			// 
			this.label1.AutoSize = true;
			this.label1.Location = new System.Drawing.Point(16, 16);
			this.label1.Name = "label1";
			this.label1.Size = new System.Drawing.Size(55, 16);
			this.label1.TabIndex = 0;
			this.label1.Text = "Bank &File:";
			// 
			// importButton
			// 
			this.importButton.Location = new System.Drawing.Point(192, 112);
			this.importButton.Name = "importButton";
			this.importButton.Size = new System.Drawing.Size(79, 23);
			this.importButton.TabIndex = 4;
			this.importButton.Text = "&Import";
			this.importButton.Click += new System.EventHandler(this.importButton_Click);
			// 
			// logTextBox
			// 
			this.logTextBox.Anchor = ((System.Windows.Forms.AnchorStyles)((((System.Windows.Forms.AnchorStyles.Top | System.Windows.Forms.AnchorStyles.Bottom) 
				| System.Windows.Forms.AnchorStyles.Left) 
				| System.Windows.Forms.AnchorStyles.Right)));
			this.logTextBox.BackColor = System.Drawing.SystemColors.Window;
			this.logTextBox.Location = new System.Drawing.Point(16, 152);
			this.logTextBox.Multiline = true;
			this.logTextBox.Name = "logTextBox";
			this.logTextBox.ReadOnly = true;
			this.logTextBox.ScrollBars = System.Windows.Forms.ScrollBars.Both;
			this.logTextBox.Size = new System.Drawing.Size(420, 209);
			this.logTextBox.TabIndex = 6;
			this.logTextBox.Text = "";
			// 
			// databaseTextBox
			// 
			this.databaseTextBox.Anchor = ((System.Windows.Forms.AnchorStyles)(((System.Windows.Forms.AnchorStyles.Top | System.Windows.Forms.AnchorStyles.Left) 
				| System.Windows.Forms.AnchorStyles.Right)));
			this.databaseTextBox.Location = new System.Drawing.Point(120, 48);
			this.databaseTextBox.Name = "databaseTextBox";
			this.databaseTextBox.Size = new System.Drawing.Size(316, 20);
			this.databaseTextBox.TabIndex = 3;
			this.databaseTextBox.Text = "";
			// 
			// browseButton
			// 
			this.browseButton.Anchor = ((System.Windows.Forms.AnchorStyles)((System.Windows.Forms.AnchorStyles.Top | System.Windows.Forms.AnchorStyles.Right)));
			this.browseButton.Location = new System.Drawing.Point(408, 16);
			this.browseButton.Name = "browseButton";
			this.browseButton.Size = new System.Drawing.Size(28, 23);
			this.browseButton.TabIndex = 9;
			this.browseButton.Text = "...";
			this.browseButton.Click += new System.EventHandler(this.browseButton_Click);
			// 
			// dataFileTextBox
			// 
			this.dataFileTextBox.Anchor = ((System.Windows.Forms.AnchorStyles)(((System.Windows.Forms.AnchorStyles.Top | System.Windows.Forms.AnchorStyles.Left) 
				| System.Windows.Forms.AnchorStyles.Right)));
			this.dataFileTextBox.Location = new System.Drawing.Point(120, 16);
			this.dataFileTextBox.Name = "dataFileTextBox";
			this.dataFileTextBox.Size = new System.Drawing.Size(284, 20);
			this.dataFileTextBox.TabIndex = 1;
			this.dataFileTextBox.Text = "";
			// 
			// runStoredProcCheckBox
			// 
			this.runStoredProcCheckBox.Checked = true;
			this.runStoredProcCheckBox.CheckState = System.Windows.Forms.CheckState.Checked;
			this.runStoredProcCheckBox.Location = new System.Drawing.Point(120, 72);
			this.runStoredProcCheckBox.Name = "runStoredProcCheckBox";
			this.runStoredProcCheckBox.Size = new System.Drawing.Size(312, 24);
			this.runStoredProcCheckBox.TabIndex = 10;
			this.runStoredProcCheckBox.Text = "&Update transactions";
			// 
			// ImportMainForm
			// 
			this.AutoScaleBaseSize = new System.Drawing.Size(5, 13);
			this.ClientSize = new System.Drawing.Size(452, 374);
			this.Controls.Add(this.runStoredProcCheckBox);
			this.Controls.Add(this.label3);
			this.Controls.Add(this.label2);
			this.Controls.Add(this.label1);
			this.Controls.Add(this.logTextBox);
			this.Controls.Add(this.databaseTextBox);
			this.Controls.Add(this.dataFileTextBox);
			this.Controls.Add(this.importButton);
			this.Controls.Add(this.browseButton);
			this.Name = "ImportMainForm";
			this.Text = "Manual Import";
			this.Load += new System.EventHandler(this.ImportMainForm_Load);
			this.ResumeLayout(false);

		}
		#endregion

		private void ImportMainForm_Load(object sender, System.EventArgs e)
		{
			databaseTextBox.Text = ImportMain.ConnectionString;

			System.Diagnostics.Trace.Listeners.Add(new TextBoxTraceListener(logTextBox));
		}

		private void browseButton_Click(object sender, System.EventArgs e)
		{
			using (OpenFileDialog dialog = new OpenFileDialog())
			{
				dialog.InitialDirectory = ImportMain.FileLocation;

				if (dialog.ShowDialog(this) == DialogResult.OK)
				{
					dataFileTextBox.Text = dialog.FileName;
				}
			}		
		}

		private void importButton_Click(object sender, System.EventArgs e)
		{
			ImportMain.ImportFile(dataFileTextBox.Text, databaseTextBox.Text);

			if (runStoredProcCheckBox.Checked)
				ImportMain.ExecuteAfterImportStoredProc(databaseTextBox.Text);
		}
	}
}